vector-stores/spring-ai-gemfire/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -311,26 +363,41 @@ public void add(List<Document> documents) {
 		}).toList());
 
 		ObjectMapper objectMapper = new ObjectMapper();
+		String embeddingsJson = null;
 		try {
-			String embeddingsJson = objectMapper.writeValueAsString(upload).substring("{\"embeddings\":".length());
+			embeddingsJson = objectMapper.writeValueAsString(upload).substring("{\"embeddings\":".length());
+		}
+		catch (JsonProcessingException e) {
+			throw new RuntimeException("Embedding json parsing error :" + e.getMessage());
+		}
 
-			client.post()
-				.uri("/" + INDEX_NAME + "/embeddings")
-				.contentType(MediaType.APPLICATION_JSON)
-				.bodyValue(embeddingsJson)
-				.retrieve()
-				.bodyToMono(Void.class)
-				.block();
+		client.post()
+			.uri("/" + INDEX_NAME + "/embeddings")
+			.contentType(MediaType.APPLICATION_JSON)
+			.bodyValue(embeddingsJson)
+			.retrieve()
+			.bodyToMono(Void.class)
+			.onErrorMap(WebClientException.class, this::handleHttpClientException)
+			.block();
+	}
+
+	private Throwable handleHttpClientException(Throwable ex) {
+		if (!(ex instanceof WebClientResponseException)) {
+			logger.warn("Got an unexpected error: {}", ex.toString());
+			return ex;
```

Comment:
What;s the difference in return vs throwing the exception?  This method does both, should they all throw or all return or is the difference intended?